### ISSUE TITLE
Building tests fails on Clang 3.6.2

### ISSUE THREAD

**Issue by offa** [cpputest/cpputest#710]
After checking building of current snapshot (b8f5ce0) with clang (3.6.2) for #709; it seems clang fails to build the `UtestTest`: 

```
cpputest/tests/UtestTest.cpp:88:12: error: comparison of
      address of 'this->fixture' not equal to a null pointer is always true
      [-Werror,-Wtautological-pointer-compare]
    CHECK(&fixture != 0);
           ^~~~~~~    ~
```

This applies to C++11 enabled and disabled.

**Comment by arstrube** [comment]
As far as I know, this code has been around for a long time. I have seen this error before, but can't remember where, unfortunately.

**Comment by offa** [comment]
Fixed by #716.
